JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,309 @@
-            ResultSet rs = executeQuery(conn);
-            printResults(rs);
-            conn.close();
-        } catch (SQLTransientConnectionException connException) {
-            System.out.print("Network connection issue: ");
-            System.out.print(connException.getMessage());
-            System.out.println(" Try again later!");
-        } catch (SQLInvalidAuthorizationSpecException authException) {
-            System.out.print("Could not log into database: ");
-            System.out.print(authException.getMessage());
-            System.out.println(" Check the login credentials and try again.");
-        } catch (SQLException e) {
-            e.printStackTrace();
-        }
-        return 0;
-    }
-
-    public static void main(String[] args) {
-        int exitCode = new CommandLine(new OAuthSampleApp()).execute(args);
-        System.exit(exitCode);
-    }
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.vertica.jdbc.VerticaConnection;
+
+public class OAuthSampleApp {
+	private static Properties prop;
+	private static Map<String, String> csProp = new HashMap<String, String>();
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+	private static String OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+	private static Connection vConnection;
+
+	// Get jdbc connection string from provided configuration
+	private static void getcsProp(String args) {
+		String[] entries = args.split(";");
+		for (String entry : entries) {
+			String[] pair = entry.split("=");
+			csProp.put(pair[0], pair[1]);
+		}
+	}
+
+	private static String getConnectionString() {
+		return "jdbc:vertica://" + csProp.get("Host") + ":" + csProp.get("Port") + "/" + csProp.get("Database")
+				+ "?user=" + csProp.get("User") + "&password=" + csProp.get("Password");
+	}
+
+	// Get the parameters from the connection String
+	private static String getParam(String paramName) {
+		return csProp.get(paramName);
+	}
+
+	// Add/Create Authentication record in database. Create User and grant the
+	// permissions for User
+	private static void SetUpDbForOAuth() throws Exception {
+		String connectionString = getConnectionString();
+		vConnection = DriverManager.getConnection(connectionString);
+		String USER = prop.getProperty("User");
+		String CLIENT_ID = prop.getProperty("ClientId");
+		String CLIENT_SECRET = prop.getProperty("ClientSecret");
+		String DISCOVERY_URL = prop.getProperty("DiscoveryUrl");
+		Statement st = vConnection.createStatement();
+		st.execute("DROP USER IF EXISTS " + USER + " CASCADE;");
```

Comment:
Can you put all of these commands into an array of strings, then iterate through the array and call st.execute() for each element of the array. I asked Kevin to make that change in the ADO sample previously as well. It makes it easier to see everything that is being executed and it's easy to modify if need be.

Reply:
Added queries to a string array and executed the queries using iterating through the array.